modellerScipion/__init__.py

```diff
@@ -27,35 +27,41 @@
 import os, subprocess
 
 from pyworkflow.utils import yellowStr
-import pwem
-
-from pwchem import Plugin as pwchem_plugin
+import pwchem
 
 from .constants import *
 
-
 _version_ = '0.1'
 _logo = "modeller_logo.png"
 _references = ['Webb2016']
 
-MODELLER_DIC = {'name': 'modeller', 'version': '10.4', 'home': 'MODELLER_HOME'}
-
-class Plugin(pwem.Plugin):
+class Plugin(pwchem.Plugin):
     _supportedVersions = [MODELLER_DIC['version']]
 
     @classmethod
     def _defineVariables(cls):
-        """ Return and write a variable in the config file.
-        """
-        cls._defineVar("MODELLER_ENV_ACTIVATION", 'conda activate modeller-env')
+        """ Return and write a variable in the config file. """
+        cls._defineEmVar(MODELLER_DIC['home'], '{}-{}'.format(MODELLER_DIC['name'], MODELLER_DIC['version']))
 
     @classmethod
     def defineBinaries(cls, env):
+        cls.addModellerPackage(env)
+    
+    @classmethod
+    def addModellerPackage(cls, env):
+        """ This function installs Modeller package. """
         MODELLER_INSTALLED = 'modeller_installed'
 
         installationCmd = cls.getCondaActivationCmd()
-        installationCmd += ' conda create -y -c salilab -n modeller-env modeller={} &&'.format(MODELLER_DIC['version'])
-        installationCmd += ' touch %s' % MODELLER_INSTALLED
+        installationCmd += ' conda create -y -c salilab -n {} modeller={} &&'.format(cls.getEnvName(MODELLER_DIC), MODELLER_DIC['version'])
+        installationCmd += ' touch %s &&' % MODELLER_INSTALLED
+        installationCmd += ' {} && export MODELLER_CONDA_PREFIX=$CONDA_PREFIX &&'.format(cls.getEnvActivationCommand(MODELLER_DIC, condaHook=False))
```

Comment:
i am not convinced about this env variable for storing the env path. also, as it is, seems like the env path is not stored prior to its first use here, so the first time you install the plugin, it won't be printed

Reply:
Yes, i am wroking in a full modification of the installation process like in scipion-chem and scipion-chem-autodock. As soon as it is ready, i'll push it to this branch, and this variable won't be used any more